### DISCUSSION TITLE
Sustaining AzuraCast

### DISCUSSION THREAD

**Discussion by gusaus** [AzuraCast/AzuraCast#3994] [General]
This article about [CiviCRM](https://civicrm.org/blog/michaelmcandrew/paths-to-sustainability) may provide some inspiration and guidance with regards to sustaining the AzuraCast project and a core team.

One of projects referenced provides some interesting hosting models -
- Free - https://github.com/tryghost/ghost
- Fully managed - https://ghost.org/pricing/
- Self hosted /w support - https://valet.ghost.org/

These CiviCRM sustainability initiatives, in addition to paid bounties/feature requests/bug fixes mentioned in this issue, may also work for AzuraCast - 

[Make It Happen (MIH)](https://civicrm.org/make-it-happen) - campaigns to get or support new features that you’d like to see into the project but which hasn’t been included within core development
[Partner Program](https://civicrm.org/become-a-partner) - for service providers that want to financially support the project
[Member Program](https://civicrm.org/become-a-member) - for end users that want to do the same.

Funds from most, if not all, of these models could be collected/distributed via some combo of AzuraCast's [GitHub Sponsors](https://github.com/sponsors/AzuraCast) and [Open collective](https://opencollective.com/azuracast) which are already set up.

I've been talking about these ideas w/ @SlvrEagle23 in various forums and thought it would good to open up the discussion here.

_Originally posted by @gusaus in https://github.com/AzuraCast/AzuraCast/issues/3301#issuecomment-805361778_

**Comment by gusaus** [comment]
Open Collective already provides a way to create campaigns similar to https://civicrm.org/make-it-happen. Unlike a bounty (relying on platforms/programs https://www.bountysource.com/ or https://gitcoin.co/explorer), this type of campaign could be created as a tier on https://opencollective.com/azuracast and embedded on a website. It would be totally up to AzuraCast how to define/manage such a program (i.e. you could set funding goals and only begin development if the goal is met)... the following may provide some inspiration for guidelines https://lab.civicrm.org/community/sustainability/-/wikis/Make-It-Happen

If it seems worth testing the waters, I'd be happy to help setup and manage on the AzuraCast collective and website... or using OpenProducer's website and collective. Also could try to bring in other folks that might be needed to setup and run these types of campaigns.

**Reply by gusaus** [reply]
@batsonjay What I'm mentioning here should provide a way to fund and develop any features that are preventing you from using AzuraCast...assuming your blockers are feature-related and there are enough users interested/able to fund.

We are making progress in coming up with a more efficient way to influence feature requests.  

https://features.azuracast.com/ is the current system.

That solution will probably be powered by some combo of Drupal + CiviCRM + Open Collective integration.

**Comment by gusaus** [comment]
With regards to using platforms like Github sponsors, Open Collective, and Patreon as tools to sustain AzuraCast, we might want to take a look at OBS as a model - 

**Contributing to the OBS Project** https://obsproject.com/contribute

> Thank you for your interest in supporting our work!
> 
> Open Broadcaster Software is an open source project freely available under the GPL and developed by a dedicated team of passionate individuals. Your contributions help support our developers and fund equipment purchases for easier testing and development. They also go towards software licenses and other project expenses such as website and server hosting.
> 
> It's important to us that OBS is available free without advertisements and your contributions make that possible!
> 
> If you're a developer looking to get involved with the project directly please view our contributing guidelines on GitHub or visit our Community Discord.

### Recurring Contributions
I like how they differentiate between Open Collective and Patreon

Open Collective - https://opencollective.com/obsproject/contribute

> Open Collective contributions go to a communal fund used to support core members of the OBS developer community. The usage of these funds is transparent and publicly visible.

> Companies that sponsor the OBS Project on Open Collective will be featured on this page, and Gold/Diamond/Premiere sponsors will be featured on the home page.

> If your business or content channel relies on Open Broadcaster Software as part of your source of income, we recommend sponsoring the OBS Project.

and Patreon - https://www.patreon.com/obsproject

> Hugh "Jim" Bailey is OBS Projects full-time lead developer and project maintainer. You can support his work directly by contributing on Patreon.

> Patrons at the $50 tier and above are listed on this page as a thank you for their support.

> If you're a creator or individual that relies on Open Broadcaster Software as part of your source of income, we recommend sponsoring Jim.

## Some things to consider for AzuraCast
What if we could incorporate some of those elements - 
- similar 'contribute' page on AzuraCast.com - https://obsproject.com/contribute
- Open Collective (https://opencollective.com/azuracast) - create tiers / benefits similar to https://opencollective.com/obsproject/contribute
- Patreon (https://www.patreon.com/bePatron?u=232463) - create tiers / benefits similar to https://www.patreon.com/obsproject
- GitHub (https://github.com/sponsors/AzuraCast) - maybe have tiers that match both your Open Collective and Patreon?

**Reply by gusaus** [reply]
After some nice discussion w/ @SlvrEagle23 @SC2Mitch + @Vaalyn, I have a little better feel for the core team that contributes code, docs, and helps manage the community. It also seems like access to funds would enable core contributors to spend more time on the project.

That said, it would be great if the core team (and anyone else who contributes) could chime in with regards to how they contribute and who/what they think would help sustain AzuraCast and their ability to contribute.

Thanks!

**Comment by gusaus** [comment]
I came across https://contributor.link/ while lurking in one of the Open Collective Slack channels.

> Contributor.Link strives to connect participating FOSS projects with entry-level contributors, using directories and other tools that will guide contributors to projects in need of contributors' specific talents, as well as providing guidelines for the participating projects to better onboard potential new contributors.

> We are looking for projects that need contributors! When your project participates in Contributor.Link, we will provide a connection between your project and new contributors. As Contributor.Link grows, we will help passionate contributors specify the types of work they are most interested in doing

Seems like it could provide a way to connect AzuraCast with contributors (once a few more folks provide input on project goals and needs).

**Comment by batsonjay** [comment]
Hi - I was the co-founder of Acquia, founded along with [Dries Buytaert](https://dri.es), the founder & project lead for Drupal, a big open source project with a long history of considering this same challenge.

Dries released the first Drupal code in 2001, and has led the project to becoming one of the worlds largest open source projects. Currently, the project has over 100,000 active contributors, is in it's 10th "major" release, and is used on millions of sites around the globe (most commonly in those that are more sophisticated than not (the latter more commonly using WordPress)).

We co-founded [Acquia](https://en.wikipedia.org/wiki/Acquia) to provide enterprise products & services around Drupal. The company was recently acquired by Vista Private Equity for an undisclosed (but unicorn) price. The company (and Dries) successfully navigated creating a healthy group of other companies in the ecosystem, as well as a happy growing & contributing community, and long-term sustainability for the project.

I would be happy to share what we learned. It may, or may not apply to AzuraCast. I'm a DJ with an internet radio station; Dries is not. So I'm more likely to share thoughts here than him 😂. I currently have a monthly contribution going to support AzuraCast (though for stinky reasons I'm not yet running my station on it.) I also am more able to speak to the commercial aspects of Acquia (vs. the project leadership / funding aspects), though (a) they overlap a lot, and (b) whatever I can't answer, I can go ask Dries (since we are still close, despite me no longer being CEO of Acquia).

 I don't know where to start with such info-sharing. Doing it on this page seems inefficient because ... e.g. where to start. Would key contributors & community members like to jump into a Zoom call or Clubhouse call (or similar) at some point?

**Reply by gusaus** [reply]
Hi @batsonjay! I'm assuming you realize you're talking with the same person behind some related Twitter discussions? 
https://twitter.com/gusaus/status/1377453743076085771

I think it's important for @SlvrEagle23 and other core contributors drive the AzuraCast facing discussion. We're also ramping up discussions about sustaining AzuraCast https://github.com/OpenProducer/community/issues/4#issuecomment-833197801 and collaborating on related services https://github.com/OpenProducer/community/issues/16 in Github and #openproducer in https://opencollective.slack.com/

As Open Collective's nonprofit entity https://www.oscollective.org/ exists to provide tools for sustaining open source projects, contributors, and communities I'd recommend focusing collaboration in their Slack.

They also work with a few other projects, communities and orgs you might recognize! 
https://opencollective.com/acquia

#communitymedia might also be another channel of interest.

Hope to see you there!

**Comment by BusterNeece** [comment]
This discussion is happening in parallel in other places, namely the OpenCollective Slack channel, and I have trouble keeping up with the same discussion being held in multiple locations, so I'm closing this thread in favor of that one.

If anyone is interested in contributing ideas on how to help sustain AzuraCast as an open-source project, feel free to join us over at https://opencollective.slack.com/.
